packages/app-store/office365video/api/callback.ts

```diff
@@ -101,8 +103,12 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
     },
   });
 
-  const state = decodeOAuthState(req);
-  return res.redirect(
-    getSafeRedirectUrl(state?.returnTo) ?? getInstalledAppPath({ variant: "conferencing", slug: "msteams" })
-  );
+  if (completedOnboarding) {
+    const state = decodeOAuthState(req);
+    return res.redirect(
+      getSafeRedirectUrl(state?.returnTo) ?? getInstalledAppPath({ variant: "conferencing", slug: "msteams" })
```

Comment:
We can't you state.returnTo? For example,  at SSR detect if onboarding is not finished then push returnTo URL? WDYT?

Reply:
It's a bit more complicated, but in a nutshell we would have to add `{returnTo: 'our_url'}` options to this mutation [AccountDialog](https://github.com/join-com/cal-com/blob/f52afd6602c69dab263b2f7f7ececc363dfd360c/packages/app-store/office365video/components/AccountDialog.tsx#L8).

It should be based on `completedOnboarding` User's flag, so we will have to fetch this values on client side :\ I believe it's better to live it as it's implemented right now, cuz I added this flag to session (added on server side) so it will be quicker :)

Reply:
for Zoom integration we still have to keep this completedOnboarding in session or add it to this file [InstallAppButton](https://github.com/join-com/cal-com/blob/f52afd6602c69dab263b2f7f7ececc363dfd360c/packages/app-store/zoomvideo/components/InstallAppButton.tsx#L6) + update Zoom callback file